src/main/java/fact/extraction/SinglePulseExtraction.java

```diff
@@ -110,6 +111,25 @@ private void addStartSliceOffset(int[][] arr) {
         }
     }
 
+    private double[] flatten(double[][] matrix2d) {
```

Comment:
And a unit test would be tremendous. Please, make flatten 2d matrix great again!!!